### PR TITLE
feat: cleaner 'all' menu with extmarks and nice mode lists

### PR DESCRIPTION
**Pull request by willothy**
image

### CODE REVIEW

File: lua/hawtkeys/ui.lua [hunk 1]
```diff
@@ -192,7 +194,26 @@ M.show_all = function()
...
+        vim.api.nvim_buf_set_lines(
+            ResultBuf,
+            i == 1 and 0 or -1,
+            -1,
+            false,
+            { l1 }
+        )
+        -- mapping rhs as extmark so the cursor skips over it
+        vim.api.nvim_buf_set_extmark(ResultBuf, ns, i - 1, 0, {
+            virt_lines = { { { l2, "Comment" } } },
```
Comment (tris203) [inline lua/hawtkeys/ui.lua]: I dont think I like the "action" being in comment highlighting, could we make this function highlighting and then make the filename in comment highlighting?

**Review by tris203 — CHANGES_REQUESTED** [review]

**Comment by tris203** [comment]
This looks great @willothy !

Thank you, do you think there is any reason that we cant use this as a general theme accross the whole plugin?

**Review by tris203 — APPROVED** [review]

**Comment by willothy** [comment]
> This looks great @willothy !
> 
> Thank you, do you think there is any reason that we cant use this as a general theme accross the whole plugin?

Thanks! I can do some work with extmarks and highlighting for the rest of the UI, and try to unify things a bit. For example, I'd like to use overlay extmarks for the search character highlights instead of `[char]`

**Comment by tris203** [comment]
> > This looks great @willothy !
> > Thank you, do you think there is any reason that we cant use this as a general theme accross the whole plugin?
> 
> Thanks! I can do some work with extmarks and highlighting for the rest of the UI, and try to unify things a bit. For example, I'd like to use overlay extmarks for the search character highlights instead of `[char]`

would you like to do that in this PR or a seperate one? @willothy

**Comment by willothy** [comment]
Probably a separate one, since this code likely won't be changed in it

**Pull request merged**